src/main/java/flink/graphs/example/MusicProfiles.java

```diff
@@ -0,0 +1,181 @@
+import java.util.List;
+import java.util.Random;
+
+import org.apache.flink.api.common.ProgramDescription;
+import org.apache.flink.api.common.functions.CoGroupFunction;
+import org.apache.flink.api.common.functions.FilterFunction;
+import org.apache.flink.api.common.functions.GroupReduceFunction;
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple1;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.types.NullValue;
+import org.apache.flink.util.Collector;
+
+import flink.graphs.Edge;
+import flink.graphs.EdgeDirection;
+import flink.graphs.EdgesFunctionWithVertexValue;
+import flink.graphs.Graph;
+import flink.graphs.Vertex;
+import flink.graphs.example.utils.MusicProfilesData;
+import flink.graphs.library.LabelPropagation;
+import flink.graphs.utils.Tuple3ToEdgeMap;
+
+public class MusicProfiles implements ProgramDescription {
+
+	/**
+	 * This example demonstrates how to mix the "record" Flink API with the graph API.
+	 * The input is a set <userId - songId - playCount> triplets and a set of
+	 * bad records,i.e. song ids that should not be trusted.
+	 * Initially, we use the record API to filter out the bad records.
+	 * Then, we use the graph API to create a user -> song weighted bipartite graph
+	 * and compute the top song (most listened) per user.
+	 * Then, we use the record API again, to create a user-user similarity graph, 
+	 * based on common songs, where two users that listen to the same song are connected.
+	 * Finally, we use the graph API to run the label propagation community detection algorithm
+	 * on the similarity graph.
+	 */
+	public static void main (String [] args) throws Exception {
+    	
+    	ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+    	final long numberOfLabels = 3;
+    	final int numIterations = 10;
+
+    	/** 
+    	 *  Read the user-song-play triplets
+    	 *  The format is <userID>\t<songID>\t<playcount>
+    	 */
+    	DataSet<Tuple3<String, String, Integer>> triplets = MusicProfilesData.getUserSongTriplets(env);
+
+    	/**
+    	 *  Read the mismatches dataset and extract the songIDs
+    	 *  The format is "ERROR: <songID trackID> song_title"
+    	 */
+    	DataSet<Tuple1<String>> mismatches = MusicProfilesData.getMismatches(env).map(new ExtractMismatchSongIds());
+
+    	/**
+    	 *  Filter out the mismatches from the triplets dataset
+    	 */
+    	DataSet<Tuple3<String, String, Integer>> validTriplets = triplets.coGroup(mismatches)
+    			.where(1).equalTo(0).with(new FilterOutMismatches());
+
+    	/**
+    	 *  Create a user -> song weighted bipartite graph
+    	 *  where the edge weights correspond to play counts
+    	 */
+    	DataSet<Edge<String, Integer>> userSongEdges = validTriplets.map(new Tuple3ToEdgeMap<String, Integer>());
```

Comment:
Would it be good if the graph had a constructor from Tuple3 and do the conversion to the Edge by itself?